etcdserver/metrics.go

```diff
@@ -32,10 +38,46 @@ var (
 		Name: "etcdserver_proposal_failed_total",
 		Help: "The total number of failed proposals.",
 	})
+
+	fileDescriptorInUse = prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "file_descriptor_in_use_total",
+		Help: "The total number of file descriptors in use.",
+	})
+	fileDescriptorLimit = prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "file_descriptor_limit",
+		Help: "The maximal number of file descriptors.",
```

Comment:
Move to /debug/vars because the value cannot be changed if etcd doesn't expose the functionality to extern.